examples/kyb-app/src/components/organisms/UIRenderer/elements/JSONForm/JSONForm.tsx

```diff
@@ -0,0 +1,109 @@
+import { CollectionFlowContext } from '@app/domains/collection-flow/types/flow-context.types';
+
+export interface JSONFormElementBaseParams extends DataCreationParams {
+  jsonFormDefinition: RJSFSchema;
+  uiSchema?: UiSchema;
+  label?: string;
+  hint?: string;
+  description?: string;
+  documentData?: AnyObject;
+}
+
+export interface JSONFormElementParams {
+  jsonFormDefinition?: { type?: string; required?: string[] };
+}
+
+export const JSONForm: UIElementComponent<JSONFormElementParams> = ({ definition, actions }) => {
+  const { state: elementState } = useUIElementState(definition);
+  const { hidden } = useUIElementProps(definition);
+  const { formSchema, uiSchema } = useMemo(
+    () => createFormSchemaFromUIElements(definition),
+    [definition],
+  );
+  const { stateApi } = useStateManagerContext();
+
+  const { payload } = useStateManagerContext();
+  const formData = useMemo(() => createInitialFormData(definition, payload), [definition, payload]);
+
+  const formRef = useRef<any>(null);
+
+  useEffect(() => {
+    const elementValue = get(payload, definition.valueDestination);
+
+    // TO DO: ADD this logic to jmespath @blokh
+    if (definition.options?.jsonFormDefinition?.type === 'array' && Array.isArray(elementValue)) {
+      //@ts-ignore
+      set(
+        payload,
+        definition.valueDestination,
+        elementValue.map(obj => ({
+          ...obj,
+          additionalInfo: {
+            customerCompany: get(payload, 'entity.data.companyName') as string,
+            companyName: (payload as CollectionFlowContext).flowConfig.companyName,
+          },
+        })),
+      );
+    }
+  }, [payload]);
+
+  const handleArrayInputChange = useCallback(
+    (values: AnyObject[]) => {
+      if (definition.options?.jsonFormDefinition?.type === 'array') {
+        const prevContext = stateApi.getContext();
+        const currentValue = get(prevContext, definition.valueDestination);
+
+        if (Array.isArray(currentValue) && currentValue.length !== values.length) {
+          set(prevContext, definition.valueDestination, values);
+          stateApi.setContext(prevContext);
+        }
+      }
+    },
+    [definition, stateApi, payload],
+  );
+
+  const handleSubmit = useCallback(() => {}, []);
+
+  const { validationErrors } = useUIElementErrors(definition);
+
+  return hidden ? null : (
+    <div className="flex flex-col gap-2">
+      <DynamicForm
+        schema={formSchema}
+        uiSchema={uiSchema}
+        fields={jsonFormFields}
+        layouts={jsonFormLayouts}
+        formData={formData}
+        ref={formRef}
+        transformErrors={transformRJSFErrors}
+        onChange={handleArrayInputChange}
+        onSubmit={handleSubmit}
+      />
+      {validationErrors && elementState.isTouched ? (
+        <ErrorsList errors={validationErrors.map(err => err.message)} />
+      ) : null}
+    </div>
+  );
+};
```

Comment:
Remember in general only `0` renders from falsy values and that can be mitigated via `!!`.
```suggestion
  if (hidden) {
    return null;
  }

  return (
    <div className="flex flex-col gap-2">
      <DynamicForm
        schema={formSchema}
        uiSchema={uiSchema}
        fields={jsonFormFields}
        layouts={jsonFormLayouts}
        formData={formData}
        ref={formRef}
        transformErrors={transformRJSFErrors}
        onChange={handleArrayInputChange}
        onSubmit={handleSubmit}
      />
      {validationErrors && elementState.isTouched && (
        <ErrorsList errors={validationErrors.map(err => err.message)} />
      )}
    </div>
  );
};
```